### PR TITLE
Enhancement/plugin configuration

### PR DESCRIPTION
**Pull request by kevgliss**

### CODE REVIEW

File: src/dispatch/plugins/dispatch_slack/messaging.py [hunk 1]
```diff
@@ -17,95 +17,82 @@
...
+    command_messages = {
+        config.slack_command_run_workflow: {
+            "response_type": "ephemeral",
+            "text": "Opening a modal to run a workflow...",
+        },
+        config.slack_command_report_tactical: {
+            "response_type": "ephemeral",
+            "text": "Opening a dialog to write a tactical report...",
+        },
+        config.slack_command_list_incidents: {
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/messaging.py:36]:
```suggestion
        config.slack_command_list_tasks: {
```

File: src/dispatch/plugins/dispatch_slack/messaging.py [hunk 2]
```diff
@@ -17,95 +17,82 @@
...
+            "response_type": "ephemeral",
+            "text": "Opening a dialog to update participant information...",
+        },
+        config.slack_command_engage_oncall: {
+            "response_type": "ephemeral",
+            "text": "Opening a dialog to engage an oncall person...",
+        },
+        config.slack_command_list_resources: {
+            "response_type": "ephemeral",
+            "text": "Listing all incident resources...",
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/messaging.py:66]:
```suggestion
            "text": "Fetching the list of incident resources...",
```

**Review by mvilanova — CHANGES_REQUESTED** [review]
Looks great. A few questions and things that need to be changed.

File: src/dispatch/plugins/dispatch_jira/plugin.py [hunk 3]
```diff
@@ -4,24 +4,45 @@
...
-)
+
+class HostingType(str, Enum):
+    """Type of Jira deployment."""
+
+    cloud = "cloud"
+    server = "server"
+
+
+class JiraConfiguration(BaseConfigurationModel):
```
[the remark this answers is not in the document]
Comment (kevgliss) [reply src/dispatch/plugins/dispatch_jira/plugin.py:28]: These are set by the incident type plugin metadata, although I suppose we could also have a plugin default.

[the remark this answers is not in the document]
Comment (kevgliss) [reply src/dispatch/plugins/dispatch_jira/plugin.py:28]: I will add them to the config and the migration file.

[the remark this answers is not in the document]
Comment (mvilanova) [reply src/dispatch/plugins/dispatch_jira/plugin.py:28]: Yes, we need a default. Plugin metadata is optional.

**Pull request merged**